src/s1reader/s1_etad.py

```diff
@@ -0,0 +1,203 @@
+"""Wrapper of s1etad module to read S1-ETAD products."""
+import datetime
+import glob
+import os
+
+import numpy as np
+from matplotlib import pyplot as plt
+from scipy.interpolate import RectBivariateSpline
+
+try:
+    import s1etad
+except ImportError:
+    raise ImportError('Can NOT import s1etad (https://gitlab.com/s1-etad/s1-etad)!')
+
+
+
+def get_eta_correction_from_slc_burst(slc_burst, eta_dir, corr_type='sum', include_tropo=True,
+                                      resample=True, plot=False, verbose=True, unit='second'):
+    """Get the ETAD correction for one S1 burst.
+
+    Parameters
+    ----------
+    slc_burst: Sentinel1BurstSlc
+        Sentinel-1 burst object from sentinel1_reader
+    eta_dir: str
+        Sentinel-1 ETAD product directory
+    corr_type: (list of) str
+        Sentinel-1 ETAD correction type:
+        sar, atm, sum, bistatic, doppler, fmrate, geodetic, ionospheric, tropospheric
+        where sar = bistatic + doppler + fmrate
+              atm = ionospheric + tropospheric
+              sum = sar + atm + geodetic
+    resample: bool
+        resample the low resolution ETA to the full resample SLC size
+    unit: str
+        output ETA correction unit:
+        pixel, second (meter is not recommended, as time info is more consistent and universal in SAR)
+
+    Returns:
+    ----------
+    slc_rg_corr: np.ndarray in float32 in size of (lines, samples)
+        S1 ETAD correction in range   direction in the unit of meter or second
+    slc_az_corr: np.ndarray in float32 in size of (lines, samples)
+        S1 ETAD correction in azimuth direction in the unit of meter or second
+    """
+    vprint = print if verbose else lambda *args, **kwargs: None
+    if unit not in ['second', 'meter', 'pixel']:
+        raise ValueError(f'Un-recognized input unit={unit}!')
+    elif unit == 'meter':
+        meter = True     #read s1etad product in meters
+    else:
+        meter = False    #read s1etad product in seconds [recommended]
+
+    # locate / read ETA burst
+    eta_burst, eta = get_eta_burst_from_slc_burst(slc_burst, eta_dir, verbose=verbose)
+
+    # read ETA correction data
+    corr_type = corr_type.lower()
+    if isinstance(corr_type, list):
+        corr_types = list(corr_type)
+    elif corr_type == 'sar':
+        corr_types = ['bistatic', 'doppler', 'fmrate']
+    elif corr_type == 'atm':
+        corr_types = ['ionospheric', 'tropospheric']
+    else:
+        corr_types = [corr_type]
```

Comment:
- What's `corr_types` going to be when some case like `corr_type == ['sar', 'atm']`?
- Is `eta_burst.get_correction()` able to take `sum`, `sar`, or `atm` as argument for parameter `corr_type`?

Reply:
+ `corr_type == ['sar', 'atm']` will stay the same, as covered in line 59-60.

> Is eta_burst.get_correction() able to take sum, sar, or atm as argument for parameter corr_type?

Yes, that's the exactly the purpose of this part.